code/modules/awaymissions/mission_code/Cabin.dm

```diff
@@ -19,56 +19,156 @@
 
 
 
+#define SECONDS_PER_LOG 150 //How many seconds a log will burn for
 
+//Fireplaces//
 
-/*Cabin code*/
 /obj/structure/fireplace
 	name = "fireplace"
+	desc = "A large stone fireplace, warm and cozy"
+	icon = 'icons/obj/fireplace.dmi'
+	icon_state = "fireplace"
+	density = 0
+	anchored = 1
+	pixel_x = -16
+	var/wood = 0
+	var/lit = 0
+
+
+/obj/structure/fireplace/proc/try_light(obj/item/O, mob/user)
+	if (!wood)
+		user << "<span class='warning'>[src] needs some wood to burn!</span>"
+		return FALSE
+	if (lit == 1)
+		user << "<span class='warning'>It's already lit!</span>"
+		return FALSE
+
+	var/lighting_text = "<span class='notice'>[user] lights the [src] with the [O].</span>"
+	if(istype(O, /obj/item/weapon/weldingtool))
+		lighting_text = "<span class='notice'>[user] lights the [src] with the [O]. What a badass. </span>"
+	else if(istype(O, /obj/item/weapon/lighter/greyscale))
+		lighting_text = "<span class='notice'>After some fiddling, [user] manages to light the [src] with [O].</span>"
+	else if(istype(O, /obj/item/weapon/lighter))
+		lighting_text =  "<span class='rose'>With a single flick of their wrist, [user] smoothly lights the [src] with [O]. Classy.</span>"
+	else if(istype(O, /obj/item/weapon/melee/energy))
+		lighting_text = "<span class='warning'>[user] swings their [O], lighting the [src] in the proccess.</span>"
+	if(O.is_hot())
+		visible_message(lighting_text)
+		lit = 1
+		//IT'S LIT FAM
+		START_PROCESSING(SSobj, src)
+		return TRUE
+
+/obj/structure/fireplace/attackby(obj/item/T, mob/user)
+	if(istype(T,/obj/item/stack/sheet/mineral/wood))
```

Comment:
Burning paper is a good idea, given that we're actually just incrementing a timer . Can give it a smaller BURN_VALUE to compensate